### ISSUE TITLE
Safety of methods appear to be illogical

### ISSUE THREAD

**Issue by kvark** [madsmtm/objc2#685]
Looking at [MTLInstanceAccelerationStructureDescriptor](https://docs.rs/objc2-metal/0.2.2/objc2_metal/struct.MTLInstanceAccelerationStructureDescriptor.html#method.setInstanceCount), I see `setInstanceCount` being safe but `setInstanceDescriptorType` being `unsafe`. That doesn't make sense.

**Comment by madsmtm** [comment]
Yeah, the simple reason is that [`setInstanceCount` is explicitly marked as safe](https://github.com/madsmtm/objc2/blob/d04845b57166d8d941e2492f357564219874e05c/framework-crates/objc2-metal/translation-config.toml#L72), while `setInstanceDescriptorType` just... Isn't (yet).

The issue is fundamental, it's unsound for us to mark something as safe without a human in the loop (this _is_ FFI to a C-like language, and C can do anything, regardless of the signature of the function).

That said, Objective-C _is_ more constrained, and _usually_ doesn't do crazy things, so I do have a few thoughts and ideas for automatically marking certain things as safe in certain cases - just haven't gone down that route yet, my priorities have been on making the bindings correct before making them convenient ;)

**Comment by kvark** [comment]
Got it, yeah. It's unfortunate if we have to go and inspect every method for safety. Compromises the idea of auto-generated bindings to start with.
Please feel free to close if this is already tracked.

**Comment by madsmtm** [comment]
I thought I'd written it down somewhere, but no, the ideas for doing some of it automatically have only existed in my head... So I'll keep this issue open, and track it here.

**Comment by madsmtm** [comment]
Heads up: I implemented the "marking things safe automatically" idea primarily in https://github.com/madsmtm/objc2/pull/783, and have opened https://github.com/madsmtm/objc2/pull/792 to do the same for Metal. Review's welcome, though it's likely I'll be merging it rather quickly (I'm behind schedule on `v0.3.2`, and I'd like to get this out with that).
